### PR TITLE
Log streaming

### PR DESCRIPTION
**Pull request by balintzs**
Changes
=======
* Constructors/destructors for Engine and runners
    * allows stopping threads safely
* Output colors optional
* Log streaming
    * new "GET" endpoint
    * "POST" endpoint configurable to trigger only
* Superclass to contain common functionality of runners
* SWF runner using boto3 Session
* Local runner saving executions in memory
* Steps configurable to attach volumes
* Steps show container logs on error

### CODE REVIEW

File: slingshot/core/runners/base.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    async def run(self, workflow, events=None):
+        return await self.follow(
+            self.trigger(workflow, events)
+        )
+
+    async def follow(self, execution):
+        raise NotImplementedError()
+
+    def _start_decider(self):
+        """Implementation of SWF decider
```
Comment (Spedge) [inline slingshot/core/runners/base.py]: Not SWF any more, is it?

File: slingshot/core/events.py [hunk 2]
```diff
@@ -299,6 +300,8 @@ def workflow_failed(self, workflow, timestamp=None):
         self._write_content('WORKFLOW FAILED', self.Color.FAILURE)
 
     def _write_content(self, msg, color=""):
+        if not self.use_colors:
+            color = ""
         content = "{}{}{}\n".format(color, msg, self.Color.ENDC if color else '')
```
Comment (Spedge) [inline slingshot/core/events.py]: Could we not use self.use_colours as the boolean here? Especially if color is already a default of ""

File: slingshot/core/runners/base.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+    def start(self):
+        """Starts the workflow runner.
+
+        Starts two thread, the first responsible for polling decision tasks,
+        the second responsible for polling activity tasks.
+        """
+        self.log.info("STARTING...")
+        ## TODO: Find a way to garanty threads are always running
+        for thread in self._decider_threads:
```
Comment (Spedge) [inline slingshot/core/runners/base.py]: Not seeing where having two pools of threads is important...

**Comment by Spedge** [comment]
Generally LGTM, other than those two comments - but I'll be getting into the gubbinz of the engine today so perhaps there's not a huge reason to do anything with base.py

File: logging.ini [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+[handler_consoleHandler]
+class=StreamHandler
+level=INFO
+formatter=simpleFormatter
+args=(sys.stdout,)
+
+[formatter_simpleFormatter]
+format=%(asctime)s %(levelname)s [%(name)s] %(message)s
+datefmt=
```
Comment (gjtempleton) [inline logging.ini]: The ground-control logging config uses YAML, think it would be best to standardise this. (https://github.com/Skyscanner/ground-control/blob/master/groundcontrol/cli.py#L27)

Is the datefmt being declared here to make it explicit as `None`?

Comment (balintzs) [inline logging.ini]: This logging config is for debugging only. Configurable logging as a feature should be submitted in a separate PR.

File: slingshot/core/events.py (hunk 2 above)
Comment (balintzs) [inline slingshot/core/events.py]: We want to have consistent handling of `color`. Depending on its value, we want to either insert `ENDC` or not.
